Fixed #215 Added handling for token.json in get_token cmd

aaaakshat opened the pull request:
IMPORTANT NOTES (please read, then delete):

The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor.".

Please make sure to mention "#bugnum" somewhere in the description of the PR. This enables Github to link the PR to the corresponding bug.

Please also make sure to follow the [style rules](https://github.com/Cloud-CV/evalai-cli/blob/master/.github/CONTRIBUTING.md).

pushkalkatara commented:
Hi @aaaakshat Nice work, can you post screenshots describing the resolution of this issue.

aaaakshat commented:
Hi @pushkalkatara,
I've added a JSONDecodeError exception to test for simple json errors such as the file existing but being corrupt/empty.
I've also added a KeyError exception in case the "token" key is not present in the token.json file.

Below is the screenshot showing an empty token.json file as well as a corrupt token.json file.

![jsonHandling](https://user-images.githubusercontent.com/33050725/71318675-fd509100-24ce-11ea-84c7-8a8e89804e5e.png)

Ram81 commented on evalai/get_token.py:
```diff
@@ -5,6 +5,7 @@
 
 from evalai.utils.config import AUTH_TOKEN_PATH
 import json
+from json.decoder import JSONDecodeError
```
@aaaakshat can you reorder the imports.

Can you move `import json` to line 2

aaaakshat replied:
Hey @Ram81 I've done so on my latest commit.

Ram81 commented on evalai/get_token.py:
```diff
@@ -1,10 +1,10 @@
-import click
 import os
+import click
```
@aaaakshat can you reorder this lexicographically? import click should be the first import

aaaakshat replied:
@Ram81 done on latest commit.

pushkalkatara commented:
@aaaakshat Can you please post the screenshot of get_token command.

aaaakshat commented:
Hi @pushkalkatara, I already posted the screenshots for this command, however, I've reattached the screenshot below along with a valid case.

![image](https://user-images.githubusercontent.com/33050725/71388344-ed51c200-2632-11ea-891c-75c4658c3341.png)

![validToken](https://user-images.githubusercontent.com/33050725/71388482-84b71500-2633-11ea-9788-970a11c89f47.png)

pushkalkatara commented:
@aaaakshat Nice work. can we be more specific with the error message for the user to understand? Like,
1. When token.json is empty, we can remove displaying the DecodeError, rather display - `Token file was found empty, Please set the token using the command evalai set_token <token>`.
2. When token.json contains invalid token: `Invalid token found, please reset the token using the command evalai set_token <token>`.

Ram81 commented on evalai/get_token.py:
```diff
@@ -29,4 +29,8 @@ def get_token():
-                echo(e)
+                echo(str(e))
+            except JSONDecodeError as e:
+                echo("Token file was found empty, Please set the token using the command evalai set_token <token>")
```
@aaaakshat can you replace `Please` with `please`

aaaakshat replied:
done.
